pallets/ethereum-events/src/lib.rs

```diff
@@ -820,7 +805,8 @@ pub mod pallet {
                 return migrations::migrate_to_multi_nft_contract::<T>()
             }
 
-            return Weight::from_ref_time(0)
+            StorageVersion::<T>::put(Releases::V4_0_0);
+            return migrations::migrate_to_bridge_contract::<T>()
```

Comment:
I think similar with some lines above, we should do a check if the storage version is already on v4 and if so, skip the migration.